src/Views/Profile/Profile.js

```diff
@@ -1,25 +1,97 @@
 // @flow
-import React, {Component} from 'react';
-import {Paper, RaisedButton} from "material-ui";
-import {PropTypes} from 'prop-types';
+import React, { Component } from 'react';
+import { Paper, RaisedButton, Avatar, FlatButton, TextField } from "material-ui";
+import { PropTypes } from 'prop-types';
+import Messages from "../../Messages";
 
 export default class Profile extends Component {
 
+    constructor(props) {
+        super(props);
+        this.handleEditClick = this.handleEditClick.bind(this);
+        this.handleSaveClick = this.handleSaveClick.bind(this);
+        this.state = { isEditable: false };
+    }
+
     static propTypes = {
         handleLogout: PropTypes.func.isRequired,
-        profile: PropTypes.object.isRequired
+        handleUpdateProfile: PropTypes.func.isRequired,
+        profile: PropTypes.object.isRequired,
     };
 
+    handleEditClick() {
+        this.setState({ isEditable: true });
+    }
+
+    handleSaveClick() {
+        this.setState({ isEditable: false });
+        this.props.handleUpdateProfile(
+            this.refs.textFieldPosition.getValue(),
```

Comment:
myślę, że zamiast refów, powinien powstać komponent "formularz edycji profilu", który miałby w swoim stanie te trzy pola. wtedy obsługę zapisu też można tam zamknąć i komponent staje się reużywalny ;) refy stosuje się tylko w ostatecznych przypadkach ;) Ale pewnie póki co to nie problem ;)